Error fixes 2 testing and documentation

stephaniesaab opened the pull request:
# Description

Finished testing coverage for the entities folder. 100% Coverage of the methods and lines obtained. Also removed some redundant methods that were only used in the tests (e.g. getters and setters that were only used to test entity objects).

Fixes #84  (Error 2)

## Type of change

- [ ] Compilation error fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation
- [X] Test cases
- [X] Redundant code removal (fix code smells)
- [ ] Data storage 
- [ ] UI element

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

Commit:
Added documentation to the Customization entity class, also removed redundant code and unused paramters in methods.

Commit:
Refactored code for Customization class, wrote tests for full coverage of Customization clas and AbsentItemNameException

Commit:
Fixed documentation of DoneList class.

Commit:
Fixed documentation for Checklist class

Commit:
Finished test coverage for Item entity class. Removed redundant Unlocked method from Item and BuyItemUC.

Commit:
Finished test coverage for Item entity class. Removed redundant Unlocked method from Item and BuyItemUC.

Commit:
Finished test coverage for the ToDoList entity, removed a redundant checkUniqueName method in the ToDoList and added documentation to the ToDoList class. Full coverage for testing the ToDoList class.

Commit:
Finished full coverage for the Pet entity class. Renamed some tests so the naming is consistent in other entities test classes. Added documentation to the Pet entity class for getter and setter methods.

Commit:
Finished full test coverage of the entities. Removed some setters and constructors from the user class that were only used in tests. Also modified those tests to use the single constructor in User.

Commit:
Removed unuser methods from the entities and tests. This just removed excess code to help with the dispensables code smell.

Commit:
Changed the ArrayList to List to make it more extensible. Although there are some unused classes and methods in this file. Those will be changed or removed soon.

stephaniesaab committed:
Merge branch 'main' into error-fixes-2-testing-and-documentation

harshsbajwa commented on src/main/java/useCases/ModifyTaskUC.java:
```diff
@@ -39,8 +39,7 @@ private boolean findTask(String taskName, User user) {
         return findTask(taskName, UserUC.u());
     }
-
-    /**
```
you uncommented something, please change

harshsbajwa reviewed (changes requested):
great work, excellent documentation, please fix the comment in modifytaskuc

stephaniesaab committed:
Update ModifyTaskUC.java

stephaniesaab committed:
Update BuyHeartUC.java

stephaniesaab committed:
Update Item.java

Commit:
Arranged the packages. Integrated the changes from the previous PR with the changes to Main.

Commit:
Arranged the packages. Integrated the changes from the previous PR with the changes to Main.

Commit:
Added tests and documentation for the AdoptPetUC and the BuyHeartsUC. In the future, it would be helpful to be able to test the methods that use the UserUC singleton.

Commit:
Added tests and documentation for the BuyItemUC. In the future, it would be helpful to be able to test the methods that use the UserUC singleton.

Commit:
Added coverage and documentation to the CheckerUCTest file.

BetterthanNothin committed:
Created TaskUITest in order to test the TaskUI display.

BetterthanNothin committed:
Merge remote-tracking branch 'origin/error-fixes-2-testing-and-documentation' into error-fixes-2-testing-and-documentation

Commit:
Added one more test and coverage documentation to the PETUIDS tests.

BetterthanNothin committed:
Merge remote-tracking branch 'origin/error-fixes-2-testing-and-documentation' into error-fixes-2-testing-and-documentation

BetterthanNothin committed:
- Added documentation to all the Input Boundaries.

- Added documentation to some use cases.

- Added BuyHeartControllerTest to test BuyHeartController.

BetterthanNothin committed:
- Added tests for all the controllers.

XinyingXu committed:
Add SaveAndLoadUserUCTest

XinyingXu committed:
Add TaskUIDSTest

XinyingXu committed:
Update SaveAndLoadUserUCTest

XinyingXu committed:
Fix errors in CompleteTaskUCTest

XinyingXu reviewed (approved):
Good job on the tests, documentation and error fixes.

harshsbajwa committed:
testing for presenters, removed unused imports elsewhere, minor changes, etc.

serenahonekin reviewed (approved):
Good work on adding test cases and documentation!

e-laflamme reviewed (approved):
Looks good!

harshsbajwa committed:
tests for PetRefresher, TasklistRefresher, SeeDoneUC, and SeeToDoUC
